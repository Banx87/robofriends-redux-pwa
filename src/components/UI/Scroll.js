import React from 'react'

const Scroll = (props) => {
    return (
        <div style={{ overflowY: 'scroll', border: '2px dashed black', height: '1000px' }}>
            {props.children}
        </div>
    )
}

export default Scroll
