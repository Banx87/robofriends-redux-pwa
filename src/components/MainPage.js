import React, { Component } from 'react'
import CardList from './Card/CardList'
import SearchBox from './Card/SearchBox'
import Scroll from './UI/Scroll.js'
import ErrorBoundary from './ErrorBoundary'
import Header from './Header/Header'

import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        })
    }

    render() {
        const {onSearchChange, robots, isPending} = this.props;
       
        return (
            <div className="tc">
                <Header />
                <SearchBox searchChange={onSearchChange} />
                <Scroll>
                    {isPending ? <h1>Loading</h1> : 
                    <ErrorBoundary>
                        <CardList robots={this.filterRobots()} />
                    </ErrorBoundary>
                    }
                </Scroll>
            </div>
        )
        // return isPending ?
        //     <h1>Goofing around</h1> : (
        //         <div className="tc">
        //             <Header/>
        //             <SearchBox searchChange={onSearchChange} />
        //             <Scroll>
        //                 <CardList robots={filteredRobots} />
        //             </Scroll>
        //         </div>
        //     )
    }
}

export default MainPage;
