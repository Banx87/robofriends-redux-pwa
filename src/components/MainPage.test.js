import React from 'react'

import {shallow} from 'enzyme';
import MainPage from './MainPage';
import Scroll from './UI/Scroll';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }

    wrapper = shallow(<MainPage {...mockProps} />)
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('Filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2}/>)
    expect(wrapper.instance().filterRobots()).toEqual([]);
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }]);
})

it('Filters robots correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    }
    const filteredRobots = []
    const wrapper3 = shallow(<MainPage {...mockProps3}/>)
    expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
})

it('renders loading component when isPending is true', () => {
    const mockProps4 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: true
    }

    const wrapper4 = shallow(<Scroll {...mockProps4}/>);

    expect(wrapper4).toMatchSnapshot();
    
    
})
    


// it('expect to render MainPage component', () => {
//     const mockStore = {
//         robots: [],
//         searchField: ''
//     }
//     expect(shallow(<MainPage store ={mockStore}/>)).toMatchSnapshot();
    
// })