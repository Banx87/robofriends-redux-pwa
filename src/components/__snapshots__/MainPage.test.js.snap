// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `ShallowWrapper {}`;

exports[`renders loading component when isPending is true 1`] = `ShallowWrapper {}`;
